pkg/inject/sidecar.go

```diff
@@ -251,3 +254,34 @@ func CleanSidecar(deployment *appsv1.Deployment) {
 		}
 	}
 }
+
+// IfNeeded inject the sidecar on deployment only if it is required
+func IfNeeded(client client.Client, instance *appsv1.Deployment) (bool, error) {
```

Comment:
Can't this be tested? It would be really good to have some tests here.

Reply:
Yes, it can :)

Reply:
Done